pandas/core/aggregation.py

```diff
@@ -5,14 +5,46 @@
 
 from collections import defaultdict
 from functools import partial
-from typing import Any, DefaultDict, List, Sequence, Tuple
+from typing import Any, DefaultDict, List, Optional, Sequence, Tuple
 
 from pandas.core.dtypes.common import is_dict_like, is_list_like
 
+from pandas.core.base import SpecificationError
 import pandas.core.common as com
 from pandas.core.indexes.api import Index
 
 
+def reconstruct_func(func, *args, **kwargs):
```

Comment:
Can you annotate this signature? The return is pretty complicated so would be better to explicitly state what that is

Reply:
ok, I will try to make it optional!